robot-server/robot_server/protocols/protocol_store.py

```
@@ -199,6 +199,14 @@ def get_all(self) -> List[ProtocolResource]:
             for r in all_sql_resources
         ]
 
+    @lru_cache(maxsize=_CACHE_ENTRIES)
+    def get_all_ids(self) -> List[str]:
+        """Get all protocol ids currently saved in this store."""
+        select_ids = sqlalchemy.select(protocol_table.c.id)
+        with self._sql_engine.begin() as transaction:
+            protocol_ids = transaction.execute(select_ids).scalars().all()
+        return protocol_ids
```

Maybe have a default value in case we fail to initialize protocol_id variable because of a failed query

---

I think this question came up in another PR that I'm having trouble digging up, but I think the way that this is written now is good.

If the transaction fails, it will raise an exception out of the `with ...` block before `protocol_ids` is used. Providing a default value for `protocol_ids` would be misleading because it would never actually be returned.

---

talked irl, leaving as is so the robot server returns an error if the database transaction fails

---

Yeah I would agree with that, ignore my comment
In the case this does fail, we would return the appropriate HTTP response when handling the exception.